code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -463,5 +468,35 @@
 	piping_layer = 4
 	icon_state = "scrub_map_on-4"
 
+/obj/machinery/atmospherics/components/unary/vent_scrubber/plunger_act(obj/item/plunger/P, mob/living/user, reinforced)
+	if(!clogged)
+		return
+
+	if(welded)
+		to_chat(user, span_notice("You cannot pump the [name] if it's welded shut!."))
+		return
+
+	to_chat(user, span_notice("You begin pumping the [name] with your plunger."))
+	if(do_after(user, 60, target = src))
+		to_chat(user, span_notice("You finish pumping the [name]."))
+		clogged = FALSE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/clog(datum/source)
+	SIGNAL_HANDLER
+	RegisterSignal(src, COMSIG_PRODUCE_MOB, .proc/produce_mob)
+	clogged = TRUE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/produce_mob(datum/source, spawned_mob, living_mobs)
+	if(welded == FALSE)
+		var/mob/new_mob = new spawned_mob(src.loc)
+		living_mobs += new_mob
+		src.visible_message(span_warning("[new_mob] crawls out of [name]!"))
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/unclog()
+	clogged = FALSE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/is_clogged()
+	return clogged
```

Comment:
This getter's a bit superfluous, you can just check `clogged`.
Normally you'd only implement something like this if you planned on overriding behavior for subtypes.